### DISCUSSION TITLE
Having trouble after upgrading

### DISCUSSION THREAD

**Discussion by LawrenceGB** [run-llama/LlamaIndexTS#1428] [Q&A] [answered]
I upgraded and now getting errors ERR_REQUIRE_ESM and lamaindex was already imported.

My env is CommonJS so is there an easy fix, maybe with an await import or will I have to use an older version?

Thanks

**Comment by himself65** [comment]
what's your code?

**Reply by LawrenceGB** [reply]
I am importing and using these -

```
const {
    LlamaParseReader, Document, ImageNode, OpenAI,
    storageContextFromDefaults, PineconeVectorStore, VectorStoreIndex,
} = require("llamaindex");

```

Then getting lot's of these (llamaindex was already imported. This breaks constructor checks and will lead to issues!)

**Comment by LawrenceGB** [comment, marked as answer]
Core was just out of date :)

**Reply by himself65** [reply]
I assume this is the issue https://github.com/run-llama/LlamaIndexTS/issues/1497
